src/main/java/duke/BH.java

```diff
@@ -0,0 +1,168 @@
+                            "Now you have " + this.getListSize() + " tasks in the list" + ui.getLine());
+                } else if (s1.equals("event")) {
+                    String s = inputArray[1];
+                    String[] arr = s.split("/at");
+                    String task = arr[0];
+                    String time = arr[1];
+                    Task newTask = new Event(task, time);
+                    this.addToList(newTask);
+                    System.out.println(ui.getLine() + "Task added:" + newTask.toString() + "\n" +
+                            "Now you have " + this.getListSize() + " tasks in the list" + ui.getLine());
+                } else if (s1.equals("delete")) {
+                    int index = Integer.parseInt(inputArray[1]) - 1;
+                    System.out.println(ui.getLine() + "Okay, I have remove this task:\n" +
+                            this.deleteTask(index) + ui.getLine());
+                } else if (s1.equals("check")) {
+                    LocalDate date = LocalDate.parse(inputArray[1]);
+                    this.checkDate(date);
+                } else if (s1.equals("find")) {
+                    String[] arr = input.split(" ", 2);
+                    String word = arr[1];
+                    this.checkWord(word);
+                } else {
+                    System.out.println("Wrong input, please check again");
+                }
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println(e.getMessage());
+            throw new DukeException("Duke exception!!!");
+        }
+    }
+
+    /**
+     * check all task in the list and print out all having same date as the input
+     *
+     * @param date the date to check
+     */
+    void checkDate(LocalDate date) {
+        System.out.println(ui.getLine());
+        for (int i = 0; i < this.getListSize(); i++) {
+            Task currTask = this.list.get(i);
+            if (currTask.getDate().equals(date)) {
+                System.out.println(currTask.toString());
+            }
+        }
+        System.out.println(ui.getLine());
+    }
+
+    void checkWord(String word) {
+        System.out.println(ui.getLine());
+        for (int i = 0; i < this.getListSize(); i++) {
+            if (this.list.get(i).containsWord(word)) {
+                System.out.println(this.list.get(i).toString());
+            }
+        }
+        System.out.println(ui.getLine());
+    }
+
+    private void addToList(Task task) {
+        this.list.add(task);
+    }
+
+    private Task deleteTask(int index) {
+        Task task =  this.list.get(index);
+        this.list.remove(index);
+        return task;
+    }
+
+    private String getList() {
+        String s = "";
+        for (int i = 0; i < this.list.size(); i++) {
+            s = s + (i + 1) + ". " + list.get(i) + "\n";
+        }
+        return s;
+    }
+
+    /**
+     * Returns number of task in the list
+     *
+     * @return numer of task in the list
```

Comment:
Not sure I like this repetition in the comments, maybe you could change the @return to 
```suggestion
     * @return size of this.arraylist
```

Reply:
Nice catch! I will correct this!